test/hocon_tests.erl

```diff
@@ -267,19 +269,70 @@ merge_when_resolve_test() ->
                         b => [#{k3 => 1}, #{k2 => 2}]}},
                  hocon:binary("a=[{k1=1}] [{k2=2}],a=[{k3=1}] [{k2=2}],b=${a}")).
 
-concat_error_test_() ->
-    [ ?_assertEqual({error, {concat_error, [{x, 1}, [1, 2]]}},
+concat_error_binary_test_() ->
+    [ ?_assertEqual({error,
+                     {concat_error,
+                      <<"failed to concat [{x,1},[1,2]] in file undefined at line 1">>}},
```

Comment:
to be search friendly, we should use `key_words_for_error_message`,
e.g. `failed_to_concat [{x,1},[1,2]] in_file <filename> at_line <linenumber>"`

and `in_file <filename>` can be skipped if it's `undefined`

Reply:
changes applied